src/cocoa/toga_cocoa/widgets/table.py

```diff
@@ -1,31 +1,58 @@
+from toga.sources import to_accessor
+
 from ..libs import *
 from .base import Widget
+from .utils import TogaData, TogaIconCell
 
 
 class TogaTable(NSTableView):
     # TableDataSource methods
     @objc_method
     def numberOfRowsInTableView_(self, table) -> int:
-        return len(self.interface.data.rows) if self.interface.data else 0
+        return len(self.interface.data) if self.interface.data else 0
 
     @objc_method
     def tableView_objectValueForTableColumn_row_(self, table, column, row: int):
-        column_index = int(column.identifier)
-        return self.interface.data.item(row, column_index)
+        row = self.interface.data[row]
```

Comment:
From an interface point of view, it "smells funny" that we've gone from a couple lines to 15 to do "the same thing". We seem to be doing a _lot_ of setup and parsing to get to the `datum`, and the benefit isn't immediately obvious to me. 

Maybe there's benefit I'm missing? Going even deeper, it seems like very little of this is _specific_ to the cocoa impl, but maybe I'm missing something.

Reply:
Which 15 lines are you referring to?

Reply:
It used to be that we were merely calling 

```
column_index = int(column.identifier)
return self.interface.data.item(row, column_index)
```

Now we're doing a lot of instantiation and value-fetching for the same result? I might be misinterpreting this.

Reply:
Ah - got it. It's a case of code that has moved. The instantiation code should only be run the first time the node is displayed; and now, it's *only* executed if the node is displayed. Previously, it was executed when the node was added, regardless of whether it was displayed. It also means that all the internal cocoa logic is retained on the Cocoa delegate, instead of being spread across the widget (potentially creating some memory retention loops etc).